docs/src/main/tut/typeclasses.md

```diff
@@ -0,0 +1,73 @@
+The typeclass pattern is a technique borrowed from haskell. You can
+think of it as an "interface" in the java sense, but universally
+quantified over some type. In scala they are encoded as a trait that
+has one type variable, such as:
```

Comment:
My feeling is that it's fine to introduce readers to phrases like *universally-quantified* as long as we also provide an intuition or an informal definition as well.

In this particular case, I think something like this might work:

> You can think of it as an "interface" in the java sense, but implemented without using inheritance. In Scala, these are implemented using a trait with a type parameter (i.e. the trait is *univerally-quantified* over a type).

I'm not proposing this exact language, just sort of thinking out loud. I think it's fine to take a very iterative approach towards writing (and improving) our documentation.